### PR TITLE
Complete Configu CLI - Export command

### PR DESCRIPTION
**Pull request by RichardAkman**
Partially completes #503 by implementing export command

Usage examples:

Filter flag:
```bash
tsx packages/cli/src/run.ts eval --store store1 --set '' --schema ./tmp/schemas2/start.cfgu.yaml | tsx packages/cli/src/run.ts export --filter 'empty($.value)' --filter 'contains($.cfgu.label,"a")'
```

Template flag:
```bash
tsx packages/cli/src/run.ts eval --store store1 --set '' --schema ./tmp/schemas2/start.cfgu.yaml | tsx packages/cli/src/run.ts export --template ./tmp/template.env
```

Format flag:
```
tsx packages/cli/src/run.ts eval --store store1 --set '' --schema ./tmp/schemas2/start.cfgu.yaml | tsx packages/cli/src/run.ts export --format 'Dotenv({json:{{pipe}},wrap:true})'
```

Schema used in example:
```yaml
keys:
  GREETING:
    default: hello
  SUBJECT:
    default: world
    label:
      - 'a'
  MESSAGE:
    description: Generates a full greeting message
```

Template used example:
```Dotenv
SUBJECT={{SUBJECT}}
GREETING={{toUpper(GREETING)}}
```

### CODE REVIEW

File: packages/cli/src/commands/export.ts [hunk 1]
```diff
@@ -19,20 +71,217 @@ export class ExportCommand extends BaseCommand {
...
+      return;
+    }
 
+    const label = this.label ?? `configs-${Date.now()}`;
+    const filter = this.filterFromFlags();
+    const keys = this.keysMutations();
+    const pipe = keys
+      ? _.mapValues(previousEvalCommandOutput, (config, key) => ({ ...config, key: keys(key) }))
+      : previousEvalCommandOutput;
+    // TODO: how do we implement the format flag? we somehow have to connect the format flag with the formatter from the registry and pass it via the 'reduce' prop of ExportCommand
```
Comment (RichardAkman) [inline packages/cli/src/commands/export.ts:280]: @davidmeirlevy, I need your input here. When providing the `format` flag, we need to fetch the formatter from the registry or download it remotely somehow but it currently doesn't expose expressions or any other logic to do this. How do you think we should implement this logic?

File: packages/cli/src/commands/export.ts [hunk 2]
```diff
@@ -19,20 +71,217 @@ export class ExportCommand extends BaseCommand {
...
+    const exportCommand = new BaseExportCommand({ pipe, filter });
+    const { result } = await exportCommand.run();
+    // TODO: the previous implementation expected the result to be an object but now export command returns a string that needs to be parsed.
```
Comment (RichardAkman) [inline packages/cli/src/commands/export.ts:283]: @davidmeirlevy, ExportCommand now returns a plain string that's reduced based on the `reduce` prop. The issue is that `exportConfigs` wants the result as a plain object for the many other flags that need to use the result such as the `template` flag. This means we will need to parse the export command result which means we now need a parser for each format which feels very redundant. How do you feel we should go about solving this? maybe we should reconsider the ExportCommand signature

File: packages/cli/src/commands/export.ts [hunk 3]
```diff
@@ -19,20 +71,217 @@ export class ExportCommand extends BaseCommand {
...
+
+      const isKeyOmitted = omittedKeys.includes(key);
+      const isLabelOmitted = _.intersection(omittedLabels, cfgu.label ?? []).length > 0;
+      const omitFilter = !isKeyOmitted && !isLabelOmitted;
+
+      return hiddenFilter && emptyFilter && pickFilter && omitFilter;
+    };
+  }
+
+  async exportConfigs(result: ExportCommandOutput, label: string) {
```
Comment (RichardAkman) [inline packages/cli/src/commands/export.ts:222]: This function needs to get the result of export command after parsing.

File: packages/cli/src/commands/export.ts [hunk 4]
```diff
@@ -19,20 +71,217 @@ export class ExportCommand extends BaseCommand {
     // ],
   });
 
-  // name = Option.String(''
+  format = Option.String('--format', {
+    description: `Format exported \`Configs\` to common configuration formats. Redirect the output to file, if needed`,
+    // TODO: get this list from somewhere? or maybe let this fail during runtime if not found in registry?
```
Comment (RichardAkman) [inline packages/cli/src/commands/export.ts:76]: @davidmeirlevy, how do you feel we should handle this?

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**